python/selection_download_sample.py

```diff
@@ -1,156 +1,156 @@
-                raise RuntimeError('Something bad happened!')
-
-        if retries == max_retries:
-            return page, api_request_time_total, disk_write_time_total
-
-        page_data = response.json()
-        page += 1
-
-        if show_progress:
-            sys.stdout.write('#')
-            sys.stdout.flush()
-
-        # accumulate data
-        if data is None:
-             data = page_data
-        else:
-             data.extend(page_data)
-
-        # dump to file every so many pages
-        if page % CHECKPOINT == 0 and data != None:
-            sys.stdout.write('+')
-            sys.stdout.flush()
-
-            start = time.time()
-
-            json.dump(data, f)
-
-            end = time.time()
-            disk_write_time_total += (end - start)
-
-            data = None
-
-        # get all pages or limit to max_pages
-        if max_pages == None:
-            pass
-        elif page == max_pages:
-            break
-
-        # have we got to the end
-        if len(page_data) < page_size:
-            break
-
-    # dump the last chunk of data, if any
-    if not data is None:
-        start = time.time()
-
-        json.dump(data, f)
-
-        end = time.time()
-        disk_write_time_total += (end - start)
-
-    return page, api_request_time_total, disk_write_time_total
-
-
-if __name__ == '__main__':
-
-    SELECTION_ID = SELECTION['ID']
-    DATA_FILE = SELECTION['NAME'] + '.tmp'
-
-    start = time.asctime(time.localtime(time.time()))
-    print(f'Started @ {start} | DATA_FILE={DATA_FILE}, PAGE_SIZE={PAGE_SIZE}, MAX_PAGES={MAX_PAGES}, CHECKPOINT={CHECKPOINT}')
-
-    with open(DATA_FILE, 'w') as f:
-        pages, api_request_time_total, disk_write_time_total = databank_download(f, SELECTION_ID)
-
-    end = time.asctime(time.localtime(time.time()))
-
-    print(f'\nCOMPLETE! Downloaded {pages} pages')
-    print(f'Started @ {start}')
-    print(f'Ended @ {end}')
-    print('Total API Request Time: %.1f secs, %.1f mins' % (api_request_time_total, api_request_time_total/60))
-    print('Total Disk Write Time: %.1f secs, %.1f mins' % (disk_write_time_total, disk_write_time_total/60))
-
-'''
-# NOTE
-# If you're using vscode to develop/debug this program and it encounters an SSL error, then do the following:
-
-#     mklink "<YOUR ANACONDA LOCATION>\Anaconda3\DLLs\libcrypto-1_1-x64.dll" "<YOUR ANACONDA LOCATION>\Anaconda3\Library\bin\libcrypto-1_1-x64.dll"
-#     mklink "<YOUR ANACONDA LOCATION>\Anaconda3\DLLs\libssl-1_1-x64.dll" "<YOUR ANACONDA LOCATION>\Anaconda3\Library\bin\libssl-1_1-x64.dll"
-
-# (or you can physically copy them over)
-
-# The exception you may see will be something like this:
-
-#     HTTPSConnectionPool(host='services.oxfordeconomics.com', port=443):
-#     Max retries exceeded with url: /api/download/...?includemetadata=false&page=0+&pagesize=2000
-#     (Caused by SSLError("Can't connect to HTTPS URL because the SSL module is not available."))
-'''
+# 
```

Comment:
Am I correct in assuming that you encountered an issue with this file claiming to be changed that you were not able to discard? I ask as I have just encountered the same when downloading the Repo and building it.

Reply:
I don't think I came across this issue. Did you manage to resolve this?